dropwizard-jooq/src/main/java/com/bendb/dropwizard/jooq/jersey/JooqBinder.java

```diff
@@ -1,35 +1,58 @@
 package com.bendb.dropwizard.jooq.jersey;
 
+import org.glassfish.hk2.api.InjectionResolver;
+import org.glassfish.hk2.api.TypeLiteral;
 import org.glassfish.hk2.utilities.binding.AbstractBinder;
 import org.glassfish.jersey.process.internal.RequestScoped;
+import org.glassfish.jersey.server.spi.internal.ValueFactoryProvider;
 import org.jooq.Configuration;
 import org.jooq.ConnectionProvider;
 import org.jooq.DSLContext;
 
+import javax.inject.Singleton;
+import java.util.Map;
+
 /**
  * Create bindings for {@link DSLContext} (via
  * {@link com.bendb.dropwizard.jooq.jersey.DSLContextFactory}),
  * {@link org.jooq.Configuration}, and {@link org.jooq.ConnectionProvider}.
+ *
+ * Also creates bindings for {@link com.bendb.dropwizard.jooq.jersey.DataSourceParam} (via
+ * {@link com.bendb.dropwizard.jooq.jersey.DataSourceParamValueFactoryProvider} and
+ * {@link com.bendb.dropwizard.jooq.jersey.DataSourceParamInjectionResolver}.
  */
 public class JooqBinder extends AbstractBinder {
 
-    private final Configuration configuration;
+    private final Map<String,Configuration> configurationMap;
+    private final Configuration defaultConfiguration;
 
-    public JooqBinder(Configuration configuration) {
-        this.configuration = configuration;
+    public JooqBinder(final Map<String,Configuration> configurationMap, final Configuration defaultConfiguration) {
+        this.configurationMap = configurationMap;
+        this.defaultConfiguration = defaultConfiguration;
     }
 
     @Override
     protected void configure() {
-        bindFactory(new DSLContextFactory(configuration))
+        final Configuration defaultConfiguration = (this.defaultConfiguration != null)
+                                                   ? this.defaultConfiguration
+                                                   : configurationMap.values().stream().findFirst().orElse(null);
+
+        bindFactory(new DSLContextFactory(defaultConfiguration))
                 .to(DSLContext.class)
                 .in(RequestScoped.class);
 
-        // Configuration and ConnectionProvider are single instance, used everywhere
+        // bind multiple instances of Configuration and ConnectionProvider
+        for (Configuration configuration : configurationMap.values()) {
+            bind(configuration).to(Configuration.class);
```

Comment:
call .named() here inside of the loop to get the @Named annotation stuff to work.

https://hk2.java.net/2.4.0-b19/xref/org/glassfish/hk2/utilities/binding/AbstractBindingBuilder.html#195